tfhe/src/shortint/mod.rs

```diff
@@ -135,3 +136,87 @@ where
 
     (cks, sks)
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub(crate) enum PaddingBit {
+    No = 0,
+    Yes = 1,
+}
+
+fn compute_delta(
+    ciphertext_modulus: CiphertextModulus,
+    message_modulus: MessageModulus,
+    carry_modulus: CarryModulus,
+    padding_bit: PaddingBit,
+) -> u64 {
+    match ciphertext_modulus.kind() {
+        CiphertextModulusKind::Native => {
+            (1u64 << (u64::BITS - 1 - padding_bit as u32)) / (carry_modulus.0 * message_modulus.0)
+                * 2
+        }
+        CiphertextModulusKind::Other | CiphertextModulusKind::NonNativePowerOfTwo => {
+            ciphertext_modulus.get_custom_modulus() as u64
+                / (carry_modulus.0 * message_modulus.0)
+                / if padding_bit == PaddingBit::Yes { 2 } else { 1 }
+                * 2
+        }
+    }
+}
+
+pub(crate) struct ShortintEncoding {
+    pub(crate) ciphertext_modulus: CiphertextModulus,
+    pub(crate) message_modulus: MessageModulus,
+    pub(crate) carry_modulus: CarryModulus,
+    pub(crate) padding_bit: PaddingBit,
+}
+
+impl ShortintEncoding {
+    pub(crate) fn delta(&self) -> u64 {
+        compute_delta(
+            self.ciphertext_modulus,
+            self.message_modulus,
+            self.carry_modulus,
+            self.padding_bit,
+        )
+    }
+}
+
+impl ShortintEncoding {
+    pub(crate) fn from_parameters(
+        params: impl Into<ShortintParameterSet>,
+        padding_bit: PaddingBit,
+    ) -> Self {
```

Comment:
When using the classical or multibit PBS, I think we should force the `PaddingBit` to be set.
But it looks like it would need bigger changes

Maybe we could think of removing no padding bit encoding and wop from shortint

Reply:
Yeah I also think the no padding bit 'support' should go away  as we don't really test it, its not something we use either